packages/react-native/src/surveys/getActiveMatchingSurveys.ts

```diff
@@ -97,23 +97,37 @@ export function getActiveMatchingSurveys(
       return true
     }
 
-    const linkedFlagCheck = survey.linked_flag_key ? flags[survey.linked_flag_key] === true : true
-    const targetingFlagCheck = survey.targeting_flag_key ? flags[survey.targeting_flag_key] === true : true
+    const linkedFlagCheck = survey.linked_flag_key ? !!flags[survey.linked_flag_key] === true : true
```

Comment:
there was a bug here, missing the `!!` operator otherwise flag strings would return false